### DISCUSSION TITLE
How to know the tera type?

### DISCUSSION THREAD

**Discussion by Kotaro-Tsukada** [hsahovic/poke-env#503] [Q&A]
`mon._terastallized_type` is usually set to None and there is no way to know the Pokémon's tera type.
Is there any way to know the tera type of a Pokémon when it is not terastallized?

**Comment by AtwMaxime** [comment]
I dont know if you're still interested, but you can easily modify poke_env to acces that information.
Go into poke_env/environment and replace the pokemon.py file with the one below.
You can now use pokemon.tera_type_2 to get the correct information without needing the pokemon to actually be terastallized.

[environment.zip](https://github.com/user-attachments/files/16714944/environment.zip)

This should fix the issue, let me know if you need anything else

**Comment by caymansimpson** [comment]
You can also access the mon's teratype through the `teampreview` field as well
